src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,123 @@
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { SortType } from '../types/sortType';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  const query = searchParams.get('query') || null;
+  const sex = searchParams.get('sex') || null;
+  const centuries = searchParams.getAll('centuries') || [];
+
+  useEffect(() => {
+    getPeople()
+      .then(setPeople)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const normalized = useCallback((currentValue: string) => {
+    return currentValue.toLowerCase().trim();
+  }, []);
+
+  const visiblePerson = useMemo(() => {
+    let temp = people.map(person => ({
+      ...person,
+      mother: people.find(mother => mother.name === person.motherName) || null,
+      father: people.find(father => father.name === person.fatherName) || null,
+    }));
+
+    if (query) {
+      temp = temp.filter(person => (
+        normalized(person.name).includes(normalized(query))
```

Comment:
you might save the `normalized(query)` into a variable to prevent doing it every time - FYI